DFT/jesus_dft.py

```diff
@@ -574,3 +575,340 @@ def show_signal_cosines(self):
         self.play(Write(original_freq_mob))
         self.play(FadeIn(times))
         self.play(LaggedStartMap(FadeIn, analysis_freqs), run_time=2)
+
+
+class IntroducePhaseProblem(MovingCameraScene):
+    def construct(self):
+
+        frame = self.camera.frame.save_state()
+        self.try_sine_wave()
+
+        self.play(*[FadeOut(mob) for mob in self.mobjects])
+        self.play(Restore(frame))
+
+        self.test_cases_again()
+
+    def try_sine_wave(self):
+        frame = self.camera.frame
+        t_min = 0
+        t_max = TAU
+
+        original_freq = 2
+
+        # samples per second
+        sample_frequency = 80
+
+        n_samples = sample_frequency
+
+        vt_frequency = ValueTracker(original_freq)
+        vt_phase = ValueTracker(0)
+        vt_amplitude = ValueTracker(1)
+        vt_b = ValueTracker(0)
+
+        def sine_cosine_redraw():
+            phase_ch_cos = get_cosine_func(
+                amplitude=vt_amplitude.get_value(),
+                freq=vt_frequency.get_value(),
+                phase=vt_phase.get_value(),
+                b=vt_b.get_value(),
+            )
+            _, phase_ch_cos_mob = plot_time_domain(phase_ch_cos, t_max=t_max)
+
+            return phase_ch_cos_mob.scale(0.6).shift(UP * 1.5)
+
+        af_matrix = get_analysis_frequency_matrix(
+            N=n_samples, sample_rate=sample_frequency, t_max=t_max
+        )
+
+        rect_scale = 0.1
+
+        def updating_transform_redraw():
+            signal_function = get_cosine_func(
+                amplitude=vt_amplitude.get_value(),
+                freq=vt_frequency.get_value(),
+                phase=vt_phase.get_value(),
+                b=vt_b.get_value(),
+            )
+
+            sampled_signal = np.array(
+                [
+                    signal_function(v)
+                    for v in np.linspace(t_min, t_max, num=n_samples, endpoint=False)
+                ]
+            ).reshape(-1, 1)
+
+            # matrix transform
+            mt = apply_matrix_transform(sampled_signal, af_matrix)
```

Comment:
Just general programming tip -- no need to comment on code that is named appropriately :) -- the function is apply_matrix_transform, no need to add a comment about it. Comments should tell readers WHY code is there.

Reply:
sometimes I forgot what ```mt``` stood for haha that's why it's there